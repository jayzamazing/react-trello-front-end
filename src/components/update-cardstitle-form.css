.updateCardsTitle {
  border: unset;
  resize: none;
  background-color: unset;
  width: 237px;
}
.updateCardsTitle:focus {
  background-color: white;
}
.update-cardstitle-area {
  background-color: white;
  border-radius: 5px;
  width: 237px;
}
.edit-cards {
  position: absolute;
  top: 0;
  right: 0;
}
