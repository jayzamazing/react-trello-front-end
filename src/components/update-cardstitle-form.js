import React from 'react';
import {reduxForm, Field, focus} from 'redux-form';
import Textarea from './textarea';
import {required, nonEmpty, length, isTrimmed} from '../validators';
import {connect} from 'react-redux';
import {updateCards} from '../actions/cards';
import {submitUpdate, blurUpdate} from './utils';
import './update-cardstitle-form.css';

export class UpdateCardsTitleForm extends React.Component {
  render() {
    return (

      <form className="update-cardstitle-area" ref="updatecardsform"
      onSubmit={
        this.props.handleSubmit(values => this.props.updateCards(this.props._id,
          values[this.props._id]._id ? values[this.props._id].title : values[this.props._id]))}>
          <Field
            format={(data) => data ? data.title : ''}
            component={Textarea}
            name={this.props._id}
            textareaClass="updateCardsTitle"
            onKeyPress={e => submitUpdate(e)}
            onBlur={() => blurUpdate(this, 'updatecardsform')}
            disabled={this.props.disabled}
          />
      </form>
    );
  }
}
const mapStateToProps = (state, props) => ({
  initialValues: state.cards
});

const mapDispatchToProps = (dispatch) => ({
  //dispatch update to cards name if enter key is pressed
  updateCards: (cardsId, cardsTitle) => {
    dispatch(updateCards(cardsId, {
      title: cardsTitle
    }));
  }
});
UpdateCardsTitleForm = reduxForm({
    enableReinitialize : true,
    onSubmitFail: (errors, dispatch) =>
        dispatch(focus('updatecards', Object.keys(errors)[0]))
})(UpdateCardsTitleForm);
export default connect(mapStateToProps, mapDispatchToProps)(UpdateCardsTitleForm);
