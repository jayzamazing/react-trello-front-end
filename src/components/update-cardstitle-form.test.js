'use strict';
import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import {UpdateCardsTitleForm} from './update-cardstitle-form';

describe('update-cardstitle-form', () => {
  //ensure that it renders without issues
  it('should render', () => {
    const initialValues = {
      "35re4qwefgt34werwd": "super testing"
    }
    const handleSubmit = jest.fn();
    shallow(<UpdateCardsTitleForm handleSubmit={handleSubmit} initialValues={initialValues} _id="35re4qwefgt34werwd"/>);
  });
  //take snapshot for future changes
  it('snapshot', () => {
    const initialValues = {
      "35re4qwefgt34werwd": "super testing"
    }
    const handleSubmit = jest.fn();
    const wrapper = shallow(<UpdateCardsTitleForm handleSubmit={handleSubmit} initialValues={initialValues} _id="35re4qwefgt34werwd"/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  //click on update button and check dispatches
  it('update cardstitle click', () => {
    const initialValues = {
      "35re4qwefgt34werwd": "super testing"
    }
    const handleSubmit = jest.fn();
    const wrapper = shallow(<UpdateCardsTitleForm handleSubmit={handleSubmit} initialValues={initialValues} _id="35re4qwefgt34werwd"/>);
    wrapper.find('[name="35re4qwefgt34werwd"]').at(0).simulate('change', { target: { value: 'testing' } });
    wrapper.find('.update-cardstitle-area').simulate('submit');
    expect(handleSubmit).toHaveBeenCalled();
  });
});
