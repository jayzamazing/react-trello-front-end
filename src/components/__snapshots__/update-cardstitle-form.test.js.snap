// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`update-cardstitle-form snapshot 1`] = `
<form
  className="update-cardstitle-area"
>
  <Field
    component={[Function]}
    format={[Function]}
    name="35re4qwefgt34werwd"
    onBlur={[Function]}
    onKeyPress={[Function]}
    textareaClass="updateCardsTitle"
  />
</form>
`;
